Adds post setup checks (#32)
HealthBioscienceIDEAS/microscopy-novice: learners/setup.md

title: Setup
---

## Download Data

> During this lessons we will be using two microscopy images and creating images

Author: Fixed 352d07409442

> of our own. Create a working directory, you can name it anything you like but

Author: Fixed 34ad001

> we recommend `microsopy-ideas`.

Author: Fixed 352d07409442

Download
 [00001_01.ome.tiff](https://downloads.openmicroscopy.org/images/OME-TIFF/2016-06/MitoCheck/00001_01.ome.tiff)
 from the open microscopy environment pages to your working directory.

Download

> [Plate1-Blue-A-12-Scene-3-P3-F2-03.czi](https://downloads.openmicroscopy.org/images/Zeiss-CZI/idr0011/Plate1-Blue-A_TS-Stinger/Plate1-Blue-A-12-Scene-3-P3-F2-03.czi)

Author: I've add links to readmes for both downloads. 3354c87

> from the open microscopy environment pages to your working dircetory.

Author: Fixed 352d07409442

## Software Setup

During these lessons we will use the [napari](https://napari.org/stable/) image
 viewer and the
[napari-aicsimagio](https://github.com/AllenCellModeling/napari-aicsimageio)
 plugin to load different image types.
For beginners we recommend using Anaconda to install and manage these
packages. The first step is to
[download and install Anaconda](https://www.anaconda.com/download#downloads).

The remaining instructions are written assuming you will be using the
Anaconda prompt. If you are unfamiliar with Anaconda it may be wise to
review the

>  [Anaconda getting started pages](https://conda.io/projects/conda/en/latest/user-guide/getting-started.html).

Author: Done 9aedfb9

> Start the Anaconda prompt and run the following commands (lines starting

Reviewer: Also, I wouldn't use 'anaconda prompt' here. Technically, the anaconda prompt is only used on windows, mac/linux just use their standard terminal

Author: 39c743c

> Start the Anaconda prompt and run the following commands (lines starting

Author: 78d0dc5

with # are comments and do not need to be run).

```bash

> # Make sure conda is upto date

Author: Fixed 352d07409442

conda update -n base conda

# We need to use pip to install the napari-aicsimagio plugin, so configure
# conda to work well with pip
conda config --set pip_interop_enabled True

# Create a virtual environment (napari-env) for the lessons
conda create -y -n napari-env -c conda-forge python=3.9 bioformats_jar

# Activate the napari-env virtual environment
conda activate napari-env

# Install napari and plugins using pip
pip install "napari[all]" napari-aicsimageio
```

## Post Setup Checks

Before starting the course it is important to check that your setup is working.

::::::::::::::::::::::::::::::::::::::: challenge

## Check napari opens a tiff image.

Any napari installation should be able to open a tiff image.

> Start napari. Note that if you've already in the napari-env virtual environment

Author: Fixed 352d07409442

Close napari: `File > exit`

::::::::::::::::::::::::::

:::::::::::::::::::: challenge

## Check napari opens a czi image.

During the course we'll be working with czi images. To open these the
napari-aicsimageio plugin is needed. Let's check the plugin is working.

Start napari. Note that if you've already done the previous test (tiff image)
 then it should not be necessary to run `conda activate napari-env`.
 However, it is necessary to restart napari to ensure the image display
 is correctly formatted.

```bash
conda activate napari-env
napari
```

`File > Open file(s)...`

- Navigate to the directory where you saved
`Plate1-Blue-A-12-Scene-3-P3-F2-03.czi` to earlier on.
- Select `Plate1-Blue-A-12-Scene-3-P3-F2-03.czi` and click open.

::::::::::::::::::::::::::
::::::::::::::::::: solution

## Expected Output

![If this is what you see your napari and aicsimageio plugin installation is working as expected. If not then please check the installation or get in touch with the course organisers for help.](fig/plate1-blue.png){alt="A screenshot of freshly opened Plate1-Blue-A-12-Scene-3-P3-F2-03.czi"

>  width='80%'}

Author: OK, I think I've shortened all lines where possible now 8cda4c7.

title: Setup
---

## Download Data

During these lessons we will be using two microscopy images and creating images
of our own. Create an empty folder somewhere on your computer to use during the
course, this will be your `working directory`. You can name it anything you
like but we recommend `microscopy-ideas`.

Download
[00001_01.ome.tiff](
https://downloads.openmicroscopy.org/images/OME-TIFF/2016-06/MitoCheck/00001_01.ome.tiff)
 ([CC BY 4.0 Licence](
https://downloads.openmicroscopy.org/images/OME-TIFF/2016-06/MitoCheck/readme.txt))
 from the open microscopy environment pages to your working directory.

Download
[Plate1-Blue-A-12-Scene-3-P3-F2-03.czi](
https://downloads.openmicroscopy.org/images/Zeiss-CZI/idr0011/Plate1-Blue-A_TS-Stinger/Plate1-Blue-A-12-Scene-3-P3-F2-03.czi)
 ([CC BY 4.0 Licence](
https://downloads.openmicroscopy.org/images/Zeiss-CZI/idr0011/readme.txt))
from the open microscopy environment pages to your working directory.

## Software Setup

During these lessons we will use the [napari](https://napari.org/stable/)
image viewer and the
[napari-aicsimagio](https://github.com/AllenCellModeling/napari-aicsimageio)

>  plugin to load different image types.

Reviewer 2: ```suggestion
 plugin to load imaging data files of various file formats.
```

Author: fixed 2b1941a

For beginners we recommend using Anaconda to install and manage these
packages. The first step is to
[download and install Anaconda](https://www.anaconda.com/download#downloads).

The remaining instructions are written assuming you will be installing and
running the software from a terminal. To open a terminal follow the
instructions for your operating system in the
['Starting Conda' section of the Anaconda getting started pages](
https://conda.io/projects/conda/en/latest/user-guide/getting-started.html#starting-conda).
Run the following commands in the terminal (lines starting
with # are comments and do not need to be run).
You can run the commands by copy and pasting them into the terminal and
pressing the <kbd>Enter</kbd>  key.

> 

Reviewer 2: What do you think about adding a statement that says "After running some of the commands, there will be a lot of output on the screen telling you the status of what the command is running. This is all normal and expected"? 
It would seem that people new to the terminal might not know what is going on at this point.

Author: I've added a "spoiler" note that people can click on to see some information about all the text on the screen. 5fd26ec

```bash
# Make sure conda is up to date
conda update -n base conda

> # We need to use pip to install the napari-aicsimagio plugin, so configure
> # conda to work well with pip

Reviewer 2: ```suggestion
# We need to use a python program called pip to install 
# the napari-aicsimagio plugin, so we need to configure
# conda to work well with pip
```
As the current target audience might not know what pip is.

Author: fixed 2b1941a

conda config --set pip_interop_enabled True

> # Create a virtual environment (napari-env) for the lessons

Reviewer 2: ```suggestion
# This line will create a "virtual environment" (called napari-env) 
# that will contain all of the software that will be used in the lessons
```

Author: fixed 2b1941a

conda create -y -n napari-env -c conda-forge python=3.9 bioformats_jar

# Activate the napari-env virtual environment
conda activate napari-env

# Install napari and plugins using pip
pip install "napari[all]" napari-aicsimageio
```

> 

Reviewer 2: These steps worked very well for me!

## Post Setup Checks

Before starting the course it is important to check that your setup is working.

::::::::::::::::::::::::::::::::::::::: challenge

## Check napari opens a tiff image.

Any napari installation should be able to open a tiff image. Start napari.
Note that if you're already in the napari-env virtual environment from the
installation steps it is not necessary to run `conda activate napari-env`.
```bash

> conda activate napari-env

Reviewer 2: Perhaps we should tell people that they should see `(napari-env)` on the left hand side of the terminal if the environment is activated?

Author: I decided not to do that as it's more information than they strictly need. Running `conda activate napari-env` from within the virtual environment has no negative effects, so they'll either know they're in the environment (more advanced users) and wonder why we keep asking them to activate it, or won't know and just keep running `conda activate napari-env` with no ill effects. 
That was my logic anyway, happy to add a note about the command prompt though if you prefer.

Reviewer 2: It was more to indicate that the activation was a success. What i had in mind is they may not know that their creation of an environment failed or not if they are not familiar with a terminal (although I guess they would get an Error message saying that environment does not exist.

Author: OK, I think that makes sense. I've added a note about the prompt in the script instructions. c4b86da

> napari

Reviewer 2: Maybe after this step, we should have them confirm that a Napari viewer should open (screenshot of a blank window), and there should not be any text in the terminal.

Author: Added screen shot 528ee63. I haven't mentioned text in terminal as some of them will get it some warnings. I get warnings about wayland and gnomes.

```
Open a file with

`File > Open file(s)...`

- Navigate to the directory where you saved `00001_01.ome.tiff` earlier on.
- Select `00001_01.ome.tiff` and click open.
- If you see a `Choose reader` dialog, select `napari builtins`.

::::::::::::::::::::::::::
::::::::::::::::::: solution

## Expected Output

![If this is what you see your napari installation is working as expected.
 If not then please check the installation or get in touch with the course
 organisers for help.](fig/ome_00001.png){alt="A screenshot of freshly
 opened 00001_01.ome.tiff" width='80%'}

> Close napari: `File > exit`

Reviewer 2: On Mac, I see `File > Close Window`

Author: Do you just get close window? I get both close window and exit (on linux). They seem to do the same thing.

::::::::::::::::::::::::::

:::::::::::::::::::: challenge

## Check napari opens a czi image.

During the course we'll be working with czi images. To open these the
napari-aicsimageio plugin is needed. Let's check the plugin is working.

Start napari. Note that if you've already done the previous test (tiff image)
then it should not be necessary to run `conda activate napari-env`.
However, it is necessary to restart napari to ensure the image display
is correctly formatted.

```bash
conda activate napari-env
napari
```

`File > Open file(s)...`

- Navigate to the directory where you saved
`Plate1-Blue-A-12-Scene-3-P3-F2-03.czi` to earlier on.

> - Select `Plate1-Blue-A-12-Scene-3-P3-F2-03.czi` and click open.

Reviewer 2: I get a dialog box where i can choose between bfio or napari-aicsimageio. Which one should I choose?

Author: I've added a note in case this happens to other users. 0eb0931

::::::::::::::::::::::::::
::::::::::::::::::: solution

## Expected Output

![If this is what you see your napari and aicsimageio plugin installation

>  is working as expected. If not then please check the installation or get

Reviewer 2: What do we mean here by "check the installation"? Can we give more details? Should they double check that each step worked?

Author: Added notes on how to check the installation. 4656fec

https://downloads.openmicroscopy.org/images/OME-TIFF/2016-06/MitoCheck/readme.txt))
 from the open microscopy environment pages to your working directory.

Download
[Plate1-Blue-A-12-Scene-3-P3-F2-03.czi](
https://downloads.openmicroscopy.org/images/Zeiss-CZI/idr0011/Plate1-Blue-A_TS-Stinger/Plate1-Blue-A-12-Scene-3-P3-F2-03.czi)
 ([CC BY 4.0 Licence](
https://downloads.openmicroscopy.org/images/Zeiss-CZI/idr0011/readme.txt))
from the open microscopy environment pages to your working directory.

## Software Setup

During these lessons we will use the [napari](https://napari.org/stable/)
image viewer and the
[napari-aicsimagio](https://github.com/AllenCellModeling/napari-aicsimageio)
 plugin to load imaging data files of various file formats.
For beginners we recommend using Anaconda to install and manage these
packages. The first step is to
[download and install Anaconda](https://www.anaconda.com/download#downloads).

The remaining instructions are written assuming you will be installing and
running the software from a terminal. To open a terminal follow the
instructions for your operating system in the
['Starting Conda' section of the Anaconda getting started pages](

> https://conda.io/projects/conda/en/latest/user-guide/getting-started.html#starting-conda).

Reviewer: Conda updated their user guide a few weeks ago. Unfortunately, this removed references to the Anaconda prompt (for Windows), instead using cmd/powershell. There are extra steps required to get conda working with these terminals, vs anaconda prompt which is a lot easier.

Instead of linking to the anaconda getting started pages, could you include the instructions for opening anaconda prompt, or a terminal for windows/mac/linux directly in the setup instructions?

Author: I've added this, let me know what you think.

Reviewer: Thanks! Could you change the Windows line to: Click Start > Search for Anaconda Prompt > Click to open

As I mentioned above, it's a lot easier to use conda on Windows with the Anaconda prompt rather than cmd/powershell

:::::::::::::::::::::::::

## Post Setup Checks

Before starting the course it is important to check that your setup is working. 

::::::::::::::::::::::::::::::::::::::: challenge

## Check napari opens a tiff image.

Any napari installation should be able to open a tiff image.

> Download [00001_01.ome.tiff](https://downloads.openmicroscopy.org/images/OME-TIFF/2016-06/MitoCheck/00001_01.ome.tiff) from the open microscopy environment pages.

Author: I've put these at the start under downloads now.

> Download [00001_01.ome.tiff](https://downloads.openmicroscopy.org/images/OME-TIFF/2016-06/MitoCheck/00001_01.ome.tiff) from the open microscopy environment pages.

Author: I suggested they create a working directory at the start. I suggested a name for or it "microsopy-ideas" but also said they can chose what they want.

Start napari
```bash
conda activate napari-env
napari
```
Open a file with

`File > Open file(s)...`

- Navigate to where you saved `00001_01.ome.tiff` to, most likely your `Downloads` folder. 
- Select `00001_01.ome.tiff` and click open. 
- If you see a `Choose reader` dialog, select `napari builtins`.

::::::::::::::::::::::::::
::::::::::::::::::: solution

## Expected Output

![If this is what you see your napari installation is working as expected. If not then please check the installation or get in touch with the course organisers for help.](fig/ome_00001.png){alt="A screenshot of freshly opened 00001_01.ome.tiff" width='80%'}

Close napari: `File > exit`

::::::::::::::::::::::::::

:::::::::::::::::::: challenge

## Check napari opens a czi image.

During the course we'll be working with czi images. To open these the napari-aicsimageio plugin is needed. Let's check the plugin is working.

> Download [Plate1-Blue-A-12-Scene-3-P3-F2-03.czi](https://downloads.openmicroscopy.org/images/Zeiss-CZI/idr0011/Plate1-Blue-A_TS-Stinger/Plate1-Blue-A-12-Scene-3-P3-F2-03.czi) from the open microscopy environment pages.

Author: I've put these under downloads at the start now.
